### PR TITLE
Integration test

### PR DESCRIPTION
**Pull request by mariuszkumor**

### CODE REVIEW

File: abixen-platform-core/src/test/java/com/abixen/platform/core/infrastructure/configuration/PlatformTestSecurityConfiguration.java [hunk 1]
```diff
@@ -14,21 +14,29 @@
...
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
 
 @Profile("test")
 @EnableWebSecurity
 @Configuration
 public class PlatformTestSecurityConfiguration extends WebSecurityConfigurerAdapter {
 
+    @Autowired
+    public void configureGlobal(AuthenticationManagerBuilder auth) throws Exception {
+        auth
```
Comment (konradkwisniewski) [inline abixen-platform-core/src/test/java/com/abixen/platform/core/infrastructure/configuration/PlatformTestSecurityConfiguration.java]: This formatting look weird.

Comment (konradkwisniewski) [reply abixen-platform-core/src/test/java/com/abixen/platform/core/infrastructure/configuration/PlatformTestSecurityConfiguration.java]: Better will by looks formating like
auth.userDetailsService(platformUserDetailsService())
       .passwordEncoder(bCryptPasswordEncoder());

File: abixen-platform-core/src/test/java/com/abixen/platform/core/infrastructure/configuration/PlatformTestSecurityConfiguration.java [hunk 2]
```diff
@@ -45,14 +53,20 @@ protected void configure(HttpSecurity http) throws Exception {
...
-                .withUser("joe.brown@abixen.com").password("password").roles("ADMIN");
+    @Bean
+    public BCryptPasswordEncoder bCryptPasswordEncoder() {
+        return new BCryptPasswordEncoder();
+    }
+
+    @Bean(name = "userDetailsService")
+    public UserDetailsService platformUserDetailsService() {
+        PlatformTestUserDetailsService platformUserDetailsService = new PlatformTestUserDetailsService();
+        return platformUserDetailsService;
```
Comment (konradkwisniewski) [inline abixen-platform-core/src/test/java/com/abixen/platform/core/infrastructure/configuration/PlatformTestSecurityConfiguration.java]: "final" missing. It is necessery to create this variable?

File: abixen-platform-core/src/test/java/com/abixen/platform/core/infrastructure/configuration/PlatformTestSecurityConfiguration.java [hunk 3]
```diff
@@ -14,21 +14,29 @@
...
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
 import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
-
-import static com.abixen.platform.common.infrastructure.util.PlatformProfiles.DEV;
-import static com.abixen.platform.common.infrastructure.util.PlatformProfiles.DOCKER;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
 
 @Profile("test")
```
Comment (konradkwisniewski) [inline abixen-platform-core/src/test/java/com/abixen/platform/core/infrastructure/configuration/PlatformTestSecurityConfiguration.java]: Why it is plain text ?. Constant will be better

File: abixen-platform-core/src/test/java/com/abixen/platform/core/infrastructure/configuration/PlatformTestUserDetailsService.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    private UserManagementService userManagementService;
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        log.debug("loadUserByUsername({})", username);
+
+        final UserDto user = userManagementService.findUser(username);
+        log.debug("user: {}", user);
+
+
```
Comment (konradkwisniewski) [inline abixen-platform-core/src/test/java/com/abixen/platform/core/infrastructure/configuration/PlatformTestUserDetailsService.java]: Should be one break line

File: abixen-platform-core/src/test/java/com/abixen/platform/core/infrastructure/configuration/PlatformTestUserDetailsService.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        if (user == null) {
+            log.error("Wrong username: {}", username);
+            throw new UsernameNotFoundException("Wrong username and / or password.");
+        }
+
+        if (!UserState.ACTIVE.equals(user.getState())) {
+            log.error("User is inactive: {}", username);
+            throw new UsernameNotFoundException("User is inactive.");
+        }
```
Comment (konradkwisniewski) [inline abixen-platform-core/src/test/java/com/abixen/platform/core/infrastructure/configuration/PlatformTestUserDetailsService.java]: this two "if" look like validation. Should be extracted to method validateUserNotNull and validateUserStateIsEqualsTo. Need to consider extracting this to validation class

File: abixen-platform-core/src/test/java/com/abixen/platform/core/infrastructure/configuration/PlatformTestUserDetailsService.java [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+        boolean enabled = true;
+        boolean accountNonExpired = true;
+        boolean credentialsNonExpired = true;
+        boolean accountNonLocked = true;
+
+        final Collection<? extends GrantedAuthority> authorities = getAuthorities(user);
+        final boolean admin = isAdmin(authorities);
+
+        final PlatformUser platformUser = new PlatformUser(user.getUsername(),
```
Comment (konradkwisniewski) [inline abixen-platform-core/src/test/java/com/abixen/platform/core/infrastructure/configuration/PlatformTestUserDetailsService.java]: Platform user with constructor (User user, other parameters) will be more readable

File: abixen-platform-core/src/test/java/com/abixen/platform/core/infrastructure/configuration/PlatformTestUserDetailsService.java [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+                user.getId(),
+                user.getSelectedLanguage(),
+                user.getAvatarFileName());
+
+        return platformUser;
+    }
+
+    private boolean isAdmin(Collection<? extends GrantedAuthority> authorities) {
+        for (GrantedAuthority grantedAuthority : authorities) {
+            if ("ROLE_ADMIN".equals(grantedAuthority.getAuthority())) {
```
Comment (konradkwisniewski) [inline abixen-platform-core/src/test/java/com/abixen/platform/core/infrastructure/configuration/PlatformTestUserDetailsService.java]: Should be constans or enum.

File: abixen-platform-core/src/test/java/com/abixen/platform/core/infrastructure/configuration/PlatformTestUserDetailsService.java [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private Collection<? extends GrantedAuthority> getAuthorities(UserDto user) {
+        List<GrantedAuthority> grantedAuthorities = new ArrayList<GrantedAuthority>();
+        for (RoleDto role : user.getRoles()) {
```
Comment (konradkwisniewski) [inline abixen-platform-core/src/test/java/com/abixen/platform/core/infrastructure/configuration/PlatformTestUserDetailsService.java]: Lambda expression will be better. 
return user.getRoles().stream
.filter()
,map()
.collect()

File: abixen-platform-core/src/test/java/com/abixen/platform/core/controller/DashboardControllerIT.groovy [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+
+        then:
+        responseEntity.getStatusCode() == HttpStatus.OK
+
+        final PageDto pageDto = responseEntity.getBody().getPage()
+        pageDto != null
+        pageDto.getId() == 1L
+        pageDto.getDescription() == "This is a sample test page"
+        pageDto.getTitle() == "Sample test"
+        pageDto.getIcon() == "fa fa-lock"
```
Comment (konradkwisniewski) [inline abixen-platform-core/src/test/java/com/abixen/platform/core/controller/DashboardControllerIT.groovy]: It is very long then. Should be decomposed on method checkPageDto, checkLayoutDto etc

File: abixen-platform-core/src/test/java/com/abixen/platform/core/infrastructure/configuration/PlatformTestSecurityConfiguration.java (hunk 3 above)
Comment (mariuszkumor) [reply abixen-platform-core/src/test/java/com/abixen/platform/core/infrastructure/configuration/PlatformTestSecurityConfiguration.java]: Intentional - I don't want to expose this profile for tests and application developers. This is used only in particular places.

File: abixen-platform-core/src/test/java/com/abixen/platform/core/infrastructure/configuration/PlatformTestSecurityConfiguration.java (hunk 1 above)
Comment (mariuszkumor) [reply abixen-platform-core/src/test/java/com/abixen/platform/core/infrastructure/configuration/PlatformTestSecurityConfiguration.java]: This is a standard of the formatting, see i.e. https://spring.io/blog/2013/07/03/spring-security-java-config-preview-web-security/

File: abixen-platform-core/src/test/java/com/abixen/platform/core/infrastructure/configuration/PlatformTestUserDetailsService.java (hunk 5 above)
Comment (mariuszkumor) [reply abixen-platform-core/src/test/java/com/abixen/platform/core/infrastructure/configuration/PlatformTestUserDetailsService.java]: It's not

File: abixen-platform-core/src/test/java/com/abixen/platform/core/infrastructure/configuration/PlatformTestUserDetailsService.java (hunk 6 above)
Comment (mariuszkumor) [reply abixen-platform-core/src/test/java/com/abixen/platform/core/infrastructure/configuration/PlatformTestUserDetailsService.java]: This is a copy from Gateway and requires more changes. Not for this commit.

File: abixen-platform-core/src/test/java/com/abixen/platform/core/infrastructure/configuration/PlatformTestUserDetailsService.java (hunk 7 above)
Comment (mariuszkumor) [reply abixen-platform-core/src/test/java/com/abixen/platform/core/infrastructure/configuration/PlatformTestUserDetailsService.java]: It's temporary likely. There is an idea to get rid of it, also from Gateway (copy).

File: abixen-platform-core/src/test/java/com/abixen/platform/core/controller/DashboardControllerIT.groovy (hunk 9 above)
Comment (mariuszkumor) [reply abixen-platform-core/src/test/java/com/abixen/platform/core/controller/DashboardControllerIT.groovy]: Let's keep the verifications in one test's method.

**Pull request merged**